tools/@aws-cdk/eslint-plugin/lib/private/import-cache.js

```diff
@@ -0,0 +1,29 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.ImportCache = void 0;
+const crypto = require("crypto");
+class ImportCache {
+    constructor() {
+        this.records = {};
+    }
+    record(record) {
+        const key = {
+            fileName: record.fileName,
+            typeName: record.typeName,
+        };
+        this.records[hashed(key)] = record;
+    }
+    find(key) {
+        return this.records[hashed(key)];
+    }
+    get imports() {
+        return Object.values(this.records);
+    }
+}
+exports.ImportCache = ImportCache;
+function hashed(key) {
+    const hash = crypto.createHash('md5');
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are using md5 which is weak hash algorithm. Cryptographic hash algorithms such as MD2, MD4, MD5, MD6, HAVAL-128, HMAC-MD5, DSA, RIPEMD, RIPEMD-128, RIPEMD-160, HMACRIPEMD160, and SHA-1 are no longer considered secure because collisions can occur. With these algorithms, it takes little computational effort to find two or more different inputs that produce the same hash. [Learn more](https://cwe.mitre.org/data/definitions/1240.html)

Reply:
fixed